2023-04-11T03:07:28.715018+0700 ERROR An error has been caught in function 'dispatch_request', process 'MainProcess' (14856), thread 'Thread-6 (process_request_thread)' (18016):
Traceback (most recent call last):

  File "C:\Python310\lib\threading.py", line 966, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x0000021DC3C3C820>
    └ <Thread(Thread-6 (process_request_thread), started daemon 18016)>

  File "C:\Python310\lib\threading.py", line 1009, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x0000021DC3C3C550>
    └ <Thread(Thread-6 (process_request_thread), started daemon 18016)>

  File "C:\Python310\lib\threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-6 (process_request_thread), started daemon 18016)>
    │    │        │    └ (<socket.socket fd=784, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 5000), raddr=...
    │    │        └ <Thread(Thread-6 (process_request_thread), started daemon 18016)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <werkzeug.serving.ThreadedWSGIServer object at 0x0000021DC5F9D390>>
    └ <Thread(Thread-6 (process_request_thread), started daemon 18016)>

  File "C:\Python310\lib\socketserver.py", line 683, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 17540)
    │    │              └ <socket.socket fd=784, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 5000), raddr=(...
    │    └ <function BaseServer.finish_request at 0x0000021DC3D63E20>
    └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000021DC5F9D390>

  File "C:\Python310\lib\socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000021DC5F9D390>
    │    │                   │        └ ('127.0.0.1', 17540)
    │    │                   └ <socket.socket fd=784, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 5000), raddr=(...
    │    └ <class 'werkzeug.serving.WSGIRequestHandler'>
    └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000021DC5F9D390>

  File "C:\Python310\lib\socketserver.py", line 747, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x0000021DC43B3520>
    └ <werkzeug.serving.WSGIRequestHandler object at 0x0000021DC5F9E440>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\serving.py", line 361, in handle
    super().handle()

  File "C:\Python310\lib\http\server.py", line 425, in handle
    self.handle_one_request()
    │    └ <function BaseHTTPRequestHandler.handle_one_request at 0x0000021DC432BE20>
    └ <werkzeug.serving.WSGIRequestHandler object at 0x0000021DC5F9E440>

  File "C:\Python310\lib\http\server.py", line 413, in handle_one_request
    method()
    └ <bound method WSGIRequestHandler.run_wsgi of <werkzeug.serving.WSGIRequestHandler object at 0x0000021DC5F9E440>>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\serving.py", line 333, in run_wsgi
    execute(self.server.app)
    │       │    │      └ <werkzeug.debug.DebuggedApplication object at 0x0000021DC5F65B10>
    │       │    └ <werkzeug.serving.ThreadedWSGIServer object at 0x0000021DC5F9D390>
    │       └ <werkzeug.serving.WSGIRequestHandler object at 0x0000021DC5F9E440>
    └ <function WSGIRequestHandler.run_wsgi.<locals>.execute at 0x0000021DC5FABAC0>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\serving.py", line 322, in execute
    for data in application_iter:
                └ <generator object DebuggedApplication.debug_application at 0x0000021DC5FC8510>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\werkzeug\debug\__init__.py", line 329, in debug_application
    app_iter = self.app(environ, start_response)
               │    │   │        └ <function WSGIRequestHandler.run_wsgi.<locals>.start_response at 0x0000021DC5FABA30>
               │    │   └ {'wsgi.version': (1, 0), 'wsgi.url_scheme': 'http', 'wsgi.input': <_io.BufferedReader name=784>, 'wsgi.errors': <_io.TextIOWr...
               │    └ <Flask 'run'>
               └ <werkzeug.debug.DebuggedApplication object at 0x0000021DC5F65B10>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 2551, in __call__
    return self.wsgi_app(environ, start_response)
           │    │        │        └ <function WSGIRequestHandler.run_wsgi.<locals>.start_response at 0x0000021DC5FABA30>
           │    │        └ {'wsgi.version': (1, 0), 'wsgi.url_scheme': 'http', 'wsgi.input': <_io.BufferedReader name=784>, 'wsgi.errors': <_io.TextIOWr...
           │    └ <function Flask.wsgi_app at 0x0000021DC4753D90>
           └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
               │    └ <function Flask.full_dispatch_request at 0x0000021DC4753250>
               └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
         │    └ <function Flask.dispatch_request at 0x0000021DC47531C0>
         └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           │    │           │    │              │    │            └ {}
           │    │           │    │              │    └ 'todoroute'
           │    │           │    │              └ <Rule '/api/todo' (DELETE, POST, OPTIONS, HEAD, GET, PUT) -> todoroute>
           │    │           │    └ {'static': <function Flask.__init__.<locals>.<lambda> at 0x0000021DC5D669E0>, 'usersroute': <function View.as_view.<locals>.v...
           │    │           └ <Flask 'run'>
           │    └ <function Flask.ensure_sync at 0x0000021DC4753520>
           └ <Flask 'run'>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask_restful\__init__.py", line 467, in wrapper
    resp = resource(*args, **kwargs)
           │         │       └ {}
           │         └ ()
           └ <function View.as_view.<locals>.view at 0x0000021DC5F5F2E0>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask\views.py", line 107, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
           │                       │    │                   └ {}
           │                       │    └ <function Resource.dispatch_request at 0x0000021DC476B130>
           │                       └ <app.routes.todo.TodoRoute object at 0x0000021DC614E110>
           └ <Flask 'run'>

> File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask_restful\__init__.py", line 582, in dispatch_request
    resp = meth(*args, **kwargs)
           │     │       └ {}
           │     └ ()
           └ <bound method TodoRoute.get of <app.routes.todo.TodoRoute object at 0x0000021DC614E110>>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask_jwt_extended\view_decorators.py", line 153, in decorator
    verify_jwt_in_request(optional, fresh, refresh, locations, verify_type)
    │                     │         │      │        │          └ True
    │                     │         │      │        └ None
    │                     │         │      └ False
    │                     │         └ False
    │                     └ False
    └ <function verify_jwt_in_request at 0x0000021DC5D49B40>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask_jwt_extended\view_decorators.py", line 89, in verify_jwt_in_request
    jwt_data, jwt_header, jwt_location = _decode_jwt_from_request(
                                         └ <function _decode_jwt_from_request at 0x0000021DC5D49FC0>

  File "D:\JOBS\pacmann\ujian\backend\venv\lib\site-packages\flask_jwt_extended\view_decorators.py", line 341, in _decode_jwt_from_request
    raise NoAuthorizationError(errors[0])
          │                    └ ['Missing Authorization Header']
          └ <class 'flask_jwt_extended.exceptions.NoAuthorizationError'>

flask_jwt_extended.exceptions.NoAuthorizationError: Missing Authorization Header
